### PR TITLE
Tbolt/2581 state switcher

### LINKED ISSUE #2581: Add ability for a user to select one of their approved/requested state affiliations
After the user has entered their MFA code, if they have previously requested or received access to multiple states, they will see a state affiliations page to select which state to view. If the user only has access to one state and doesn't have any pending requests for other states, then they will skip this page.

**Note - not needed to put the status in this page**

From the dashboard, the user can chose to switch states in the dropdown menu. This will take them back to the state affiliation selection page.

### This task is done when...
 - [ ] the State Affiliation selection form is created
 - [ ] the dropdown menu has an option to switch states
 - [ ] clicking the switch states button takes the user back to the State Affiliation selection form

### LINKED ISSUE #3205: [Tech Debt] Refactor user objects in redux to reduce duplication
We are setting the user data in two places in redux: /auth/user and /user/data objects get set with identical data. This is because LOGIN_SUCCESS reducer is in both reducers/user.js and reducers/auth.js. We should consider refactoring this to only provide one location to handle user data.

### This task is done when...
Note: "user data" in this context is referring to the response from /me which is more like "user profile/config" data. 
- [ ] Team solution for determining where user data should be stored
- [ ] Associated redux actions/reducers updated to only store user data in one location

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #2581 
Resolves #3205 

**Description-**

Adds ability to switch between different affiliations when more than one is available (either requested, approved, denied, or revoked).

- Adds menu item to select affiliation page
- Adds select affiliation page
- Allows switching between different affiliation
- On login if more than 1 affiliation is available the system will prompt to select which one to view

Test Cases
Given | When | Then | Result
------------ | ------------- | ------------ | -------------
An account that has not selected an affiliation | Logging in for the first time and requesting a single affiliation | System should redirect to the thank you page, then to the selected states dashboard with a "pending status" | 
An account that has not selected an affiliation | Logging in for the first time and requesting multiple affiliations | System should redirect to the thank you page, then to the "Select Affiliation" page | 
An account that has a previously selected single affiliation | Logging in | System should redirect to the dashboard of that single state | 
An account that has a previously selected multiple affiliations | Logging in | System should redirect the Select Affiliation page | 

Known issue(s):
* When logging with multiple state affiliations, if you select "Cancel" on the "Select Affiliation" page, you will be taken back to the login page but in a awkward state where it states "You've been securely logged out" and the logging in button is spinning. As well, reloading the page will take you to the dashboard.
* Reloading the page when on the "Select State" screen takes them to the dashboard since they're technically "logged in"
* Requesting a new affiliation will return the user back to the dashboard with the default state. 

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by thetif — APPROVED** [review]
looks good

File: web/src/reducers/user.js [hunk 1]
```diff
@@ -19,7 +23,7 @@ const user = (state = initialState, action) => {
   switch (action.type) {
     case ADMIN_EDIT_ME_REQUEST:
       return { ...state, error: false, fetching: true };
-    case LOGIN_SUCCESS:
+    case UPDATE_USER_INFO:
     case ADMIN_EDIT_ME_SUCCESS:
```
[the remark this answers is not in the document]
Comment (tbolt) [reply web/src/reducers/user.js:27]: Yea, I agree. It's currently in 18 places across 8 files. Mostly tests. I figured it best to leave it for now. Perhaps a new ticket for removing/updating this and any other admin-related pieces that are either out of date (ie. not using our actual roles) or no longer relevant.

**Comment by mirano-darren** [comment]
Good job! Passed all the test cases listed. 

|Given|When|Then|Result|
|---|---|---|---|
|An account that has not selected an affiliation | Logging in for the first time and requesting a single affiliation | System should redirect to the thank you page, then to the selected states dashboard with a "pending status" |:white_check_mark:PASS|
|An account that has not selected an affiliation | Logging in for the first time and requesting multiple affiliations | System should redirect to the thank you page, then to the "Select Affiliation" page |:white_check_mark:PASS|
|An account that has a previously selected single affiliation | Logging in | System should redirect to the dashboard of that single state |:white_check_mark:PASS|
|An account that has a previously selected multiple affiliations | Logging in | System should redirect the Select Affiliation page |:white_check_mark:PASS|

Known issues give the same result described on my local machine. Going to continue with bug testing this feature.

**Review by akuas — APPROVED** [review]
It works

**Review by jeromeleecms — APPROVED** [review]
Functions as described.

**Comment by jeromeleecms** [comment]
Also spoke with @thetif -- the sysadmin login gets stuck at the state switcher screen, she will test further and log as a bug if needed.

**Pull request merged**